api/lib/mavat/planInstructions/index.js

```diff
@@ -0,0 +1,57 @@
+const fs = require('fs');
+const util = require('util');
+const path = require('path');
+
+const { extractPdfData } = require('./parser/');
+
+async function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+const readdir = util.promisify(fs.readdir);
+const unlink = util.promisify(fs.unlink);
+
+const processPlanInstructionsFile = async (fileDir) => {
+    let files;
+    let currentTry = 0;
+
+    files = await readdir(fileDir);
+
+    while ((!files || !files.length)  && currentTry < 3){
+        // naive way to poll on file download
+        await sleep(2000);
+        files = await readdir(fileDir);
+        currentTry++;
+    }
+
+    if (files.length > 0){
```

Comment:
What happens when `files` is 0?
Should this return faise?
If this is the case it's better to return false on the reverse condition and limit the nesting of the code.